decoder/src/log.rs

```diff
@@ -42,16 +43,28 @@ pub fn log_defmt(
     let target = format!("{}{}", DEFMT_TARGET_MARKER, timestamp);
     let display = frame.display_message();
 
-    log::logger().log(
-        &Record::builder()
-            .args(format_args!("{}", display))
-            .level(level)
-            .target(&target)
-            .module_path(module_path)
-            .file(file)
-            .line(line)
-            .build(),
-    );
+    if let Some(level) = level {
+        log::logger().log(
+            &Record::builder()
+                .args(format_args!("{}", display))
+                .level(level)
+                .target(&target)
+                .module_path(module_path)
+                .file(file)
+                .line(line)
+                .build(),
+        );
+    } else {
+        log::logger().log(
+            &Record::builder()
+                .args(format_args!("{}", display))
+                .target(&target)
+                .module_path(module_path)
+                .file(file)
+                .line(line)
+                .build(),
```

Comment:
Since you don't call `.level(...)` here, which level will end up in the `Record`?

Reply:
Ah, looks like `info`: https://docs.rs/log/0.4.14/src/log/lib.rs.html#1157

Reply:
The alternative is to be more transparent, for example:
```rust
        log::logger().log(
            &Record::builder()
                .args(format_args!("{}", display))
                .level(level.unwrap_or(Level::Trace))
                .target(&target)
                .module_path(module_path)
                .file(file)
                .line(line)
                .build(),
        );
```

and remove the `if else` block completely.

Reply:
It's better to be consistent with the underlying default level in `log`, therefore the call is now `.level(level.unwrap_or(Level::Info)`